src/lib.rs

```diff
@@ -4221,6 +4270,161 @@ fn map_darwin_target_from_rust_to_compiler_architecture(target: &Target) -> &str
+                        "n" | "no" | "off" | "false" => "off",
+                        _ => return None,
+                    };
+                    Some(format!("-fembed-bitcode={cc_val}").into())
+                } else {
+                    None
+                }
+            }
+            _ if flag.starts_with("force-frame-pointers") => {
+                let force_frame_pointers = if let Some((_, rustc_val)) = flag.split_once("=") {
+                    match rustc_val {
+                        "y" | "yes" | "on" | "true" => true,
+                        "n" | "no" | "off" | "false" => false,
+                        _ => return None,
+                    }
+                } else {
+                    true
+                };
+                let cc_flag = if force_frame_pointers {
+                    "-fno-omit-frame-pointer"
+                } else {
+                    "-fomit-frame-pointer"
+                };
+                Some(cc_flag.into())
+            }
+            _ if flag.starts_with("link-dead-code") => match flag.split_once("=") {
+                Some((_, "n" | "no" | "off" | "false")) => Some("-dead_strip".into()),
+                _ => None,
+            },
+            _ if flag.starts_with("lto") => {
+                let lto_mode = if let Some((_, rustc_val)) = flag.split_once("=") {
+                    match rustc_val {
+                        "y" | "yes" | "on" | "true" | "fat" => "full",
+                        "thin" => "thin",
+                        _ => return None,
+                    }
+                } else {
+                    "full"
+                };
+
+                Some(format!("-flto={lto_mode}").into())
+            }
+            _ if flag.starts_with("no-redzone") => {
+                let no_redzone = if let Some((_, rustc_val)) = flag.split_once("=") {
+                    match rustc_val {
+                        "y" | "yes" | "on" | "true" => true,
+                        "n" | "no" | "off" | "false" => false,
+                        _ => return None,
+                    }
+                } else {
+                    true
+                };
+
+                let cc_flag = if no_redzone {
+                    "-mno-red-zone"
+                } else {
+                    "-mred-zone"
+                };
+                Some(cc_flag.into())
+            }
+            _ if flag.starts_with("relocation-model") => {
+                if let Some((_, rustc_val)) = flag.split_once("=") {
+                    let cc_flag = match rustc_val {
+                        "pic" => "-fPIC",
+                        "pie" => "-fPIE",
+                        "dynamic-no-pic" => "-mdynamic-no-pic",
+                        _ => return None,
+                    };
+                    Some(cc_flag.into())
+                } else {
+                    None
+                }
+            }
+            _ if flag.starts_with("soft-float") => {
```

Comment:
This was recently deprecated btw:
https://github.com/rust-lang/rust/issues/129893

Reply:
Ah yes you're right. Good question, on one hand it's deprecated but on the other hand if someone does pass it to rustc then they *really* need to make sure the C code is compiled with the equivalent flag too so there might be value in keeping it. I don't really mind either way so it's up to you, should I drop it?

Reply:
IMO we should support both the new `-Ctarget-feature=+soft-float` and this deprecated one, otherwise the binary generated will be broken and it'd be very very hard to track it down.

Reply:
Yeah, let's keep it, but perhaps with a comment that it's deprecated (and why we still choose to keep it).

Reply:
If someone uses the flag they'll already get the deprecation warning from rustc, but we can add a comment in the code as well I suppose